Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -99,7 +99,8 @@ class PurchasesOrchestrator {
         self.manageSubscriptionsHelper = manageSubscriptionsHelper
         self.beginRefundRequestHelper = beginRefundRequestHelper
 
-        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+           systemInfo.storeKit2Setting == .enabledForCompatibleDevices {
```

Comment:
👌🏼

Reply:
feels like we should have an accompanying test for this, so we don't inadvertently break it in the future. It can be done as a separate PR